src/parser/statement.js

```diff
@@ -692,9 +693,13 @@ pp.parseClassBody = function (node) {
     if (!method.computed) {
       let { key } = method;
 
+      if (this.hasPlugin("flow") && this.isRelational("<")) {
+        isGeneric = true;
```

Comment:
ie. you could do something like:

```javascript
pp.isClassMutatorStarter = function () {
  return doSomethingHere;
};
```

and use `this.isClassMutatorStarter()` in the `if` check and then overload `isClassMutatorStarter` to do your checking.